src/main/java/com/basho/riak/client/core/codec/TermToBinaryCodec.java

```diff
@@ -322,30 +324,37 @@ private static Cell parseCell(List<RiakTsPB.TsColumnDescription> columnDescripti
     {
         if (cell instanceof OtpErlangBinary)
         {
-            OtpErlangBinary v = (OtpErlangBinary) cell;
-            String s = new String(v.binaryValue(), StandardCharsets.UTF_8);
-            return new Cell(s);
+            final OtpErlangBinary v = (OtpErlangBinary) cell;
+            if (columnDescriptions.get(j).getType() == RiakTsPB.TsColumnType.VARCHAR)
```

Comment:
We do fall through to the "default" case on line 367 / 368 where we throw an exception, but I guess it could be more specific.